### PR TITLE
LOG-2553: Always send infra logs to infra indices

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* fixes the case where structured logging is enabled and infra source logs are sent to an application structured index
* enables watch of the api server
* disables allowing_orphan

ref: https://issues.redhat.com/browse/LOG-2553

### CODE REVIEW

**Comment by jcantrill** [comment]
likely blocked by https://github.com/openshift/cluster-logging-operator/pull/1476

File: internal/generator/vector/conf_test.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package vector
 import (
 	"encoding/json"
 	"fmt"
+	helpers2 "github.com/openshift/cluster-logging-operator/test/framework/unit"
```
Comment (cahartma) [inline internal/generator/vector/conf_test.go:6]: Should be used as `unit` below.

Comment (jcantrill) [reply internal/generator/vector/conf_test.go:6]: i don't remember doing this but I'll fix it

**Pull request merged**